### PR TITLE
feat: Refactor test suite to use deploy/init libs and point to mainnet (SC-704)

### PR DESCRIPTION
**Pull request by lucas-manuel**

### CODE REVIEW

File: test/mainnet-fork/CCTPCalls.t.sol [hunk 1]
```diff
@@ -165,59 +181,88 @@ contract BaseChainUSDCToCCTPTestBase is ForkTestBase {
...
+            freezer       : freezer,
+            relayer       : relayer,
+            oldController : address(0),  // Empty
+            psm           : address(psmBase),
+            cctpMessenger : CCTP_MESSENGER_BASE,
+            usdc          : USDC_BASE
+        });
 
-        vm.startPrank(admin);
+        RateLimitData memory usdcDepositData = RateLimitData({
```
Comment (hexonaut) [inline test/mainnet-fork/CCTPCalls.t.sol:229]: Same here these can be collapsed into 1 or 2 rate limit data structs.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Mostly looks good. Some minor changes. Assuming the PSM3 deposit limits will be updated in another PR based on our discussions.

File: test/mainnet-fork/ForkTestBase.t.sol [hunk 2]
```diff
@@ -192,88 +176,107 @@ contract ForkTestBase is DssTest {
...
 
-        mainnetController.grantRole(FREEZER, freezer);
-        mainnetController.grantRole(RELAYER, relayer);
+        RateLimitData memory usdcToUsdsData = RateLimitData({
+            maxAmount : 5_000_000e6,
+            slope     : uint256(1_000_000e6) / 4 hours
+        });
 
-        almProxy.grantRole(CONTROLLER, address(mainnetController));
+        RateLimitData memory usdcToCctpData = RateLimitData({
```
Comment (hexonaut) [inline test/mainnet-fork/ForkTestBase.t.sol:236]: This one can be unlimited. I think that is more likely in practise.

File: test/base-fork/ForkTestBase.t.sol [hunk 3]
```diff
@@ -58,80 +70,100 @@ contract ForkTestBase is Test {
...
+            admin         : SPARK_EXECUTOR,
+            freezer       : freezer,
+            relayer       : relayer,
+            oldController : address(0),  // Empty
+            psm           : address(psmBase),
+            cctpMessenger : CCTP_MESSENGER_BASE,
+            usdc          : USDC_BASE
+        });
+
+        RateLimitData memory usdcDepositData = RateLimitData({
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/base-fork/ForkTestBase.t.sol:129]: Updated

File: test/base-fork/ForkTestBase.t.sol [hunk 4]
```diff
@@ -58,80 +70,100 @@ contract ForkTestBase is Test {
...
 
-        foreignController.grantRole(FREEZER, freezer);
-        foreignController.grantRole(RELAYER, relayer);
+        RateLimitData memory usdcWithdrawData = RateLimitData({
+            maxAmount : 5_000_000e6,
+            slope     : uint256(1_000_000e6) / 4 hours
+        });
 
-        almProxy.grantRole(CONTROLLER, address(foreignController));
+        RateLimitData memory usdcToCctpData = RateLimitData({
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/base-fork/ForkTestBase.t.sol:139]: Updated

File: test/mainnet-fork/CCTPCalls.t.sol (hunk 1 above)
Comment (lucas-manuel) [reply test/mainnet-fork/CCTPCalls.t.sol:229]: Updated

File: test/mainnet-fork/ForkTestBase.t.sol [hunk 5]
```diff
@@ -192,88 +176,107 @@ contract ForkTestBase is DssTest {
...
 
-        IVaultLike(ilkInst.vault).rely(address(almProxy));
+        RateLimitData memory usdsMintData = RateLimitData({
+            maxAmount : 5_000_000e18,
+            slope     : uint256(1_000_000e18) / 4 hours
+        });
 
-        mainnetController.grantRole(FREEZER, freezer);
-        mainnetController.grantRole(RELAYER, relayer);
+        RateLimitData memory usdcToUsdsData = RateLimitData({
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/mainnet-fork/ForkTestBase.t.sol:231]: Updated

File: test/mainnet-fork/ForkTestBase.t.sol (hunk 2 above)
Comment (lucas-manuel) [reply test/mainnet-fork/ForkTestBase.t.sol:236]: Updated

File: test/base-fork/ForkTestBase.t.sol (hunk 4 above)
[the remark this answers is not in the document]
Comment (hexonaut) [reply test/base-fork/ForkTestBase.t.sol]: It's still limited.

File: test/mainnet-fork/ForkTestBase.t.sol (hunk 2 above)
Comment (hexonaut) [reply test/mainnet-fork/ForkTestBase.t.sol]: It's still limited.

**Review by hexonaut — APPROVED** [review]

**Pull request merged**